contracts/TransferController.sol

```diff
@@ -30,11 +31,29 @@ contract TransferController {
     function isTyped() public view returns(bool);
     function isQuantifiable() public view returns(bool);
 
+    /// @notice MUST be called with DELEGATECALL
     function receive(address from, address to, uint256 amount, address currencyCt, uint256 currencyId) public;
 
     /// @notice MUST be called with DELEGATECALL
     function approve(address to, uint256 amount, address currencyCt, uint256 currencyId) public;
 
     /// @notice MUST be called with DELEGATECALL
-    function send(address to, uint256 amount, address currencyCt, uint256 currencyId) public;
+    function send(address from, address to, uint256 amount, address currencyCt, uint256 currencyId) public;
+
+    //----------------------------------------
+
+    /// @notice KEEP as `view` because `pure` raises revert
+    function getApproveSignature() public view returns (bytes4) {
+        return bytes4(keccak256("approve(address,uint256,address,uint256)"));
```

Comment:
As this function no longer makes us of the state constants commented out above the function could be declared as `pure` rather than `view`. And idem with the two functions below.